internal/infrastructure/db/migrations/20231113225121_create_template_dd.sql

```diff
@@ -0,0 +1,29 @@
+-- +goose Up
+-- +goose StatementBegin
+SELECT 'up SQL query';
+
+CREATE TABLE IF NOT EXISTS template_device_definitions (
+    id BIGSERIAL,
+    device_definition_id BIGINT NOT NULL,
+    template_name TEXT REFERENCES templates(template_name) NOT NULL,
```

Comment:
we should probably also add `device_style_id` as a nullable property, for cases where we want to provide an even more granular template to a specific Style trim, eg. HEV vs. ICE.  This matches with the device_styles table in the device_definitions repo, also ksuid